src/main/java/com/mcmoddev/alt/util/ALTFileUtils.java

```diff
@@ -113,4 +120,77 @@ public static void safeDeleteDirectory(File file) {
 		}
 	}
 
+	public static void copyFromResourceIfNotPresent(ResourceLocation value) {
+		Path base = Paths.get(AdditionalLootTables.getLootFolder().toString(), value.getResourceDomain()).normalize();
+		createDirectoryIfNotPresent( base );
+		File container = Loader.instance().getIndexedModList().get(value.getResourceDomain()).getSource();
+		Path root = container.toPath().resolve(Paths.get("assets", value.getResourceDomain(), value.getResourcePath()));
+		
+		if( !root.toFile().isDirectory() ) {
+			AdditionalLootTables.logger.error("Mod {} asked us to load from {} but it is not a directory!", value.getResourceDomain(), root.toString());
+			return;
+		}
+		
+		
+		Iterator<Path> itr = null;
+			
+		try {
+			itr = Files.walk(root).iterator();				
+		} catch( IOException e) {
+			AdditionalLootTables.logger.error("Getting iterator for resource of mod {}", value.getResourceDomain(), e);
+			return;
+		}
+		
+		while( itr != null && itr.hasNext() ) {
+			Path current = itr.next();
+				
+			if( current.toFile().isDirectory() ) {

```

Comment:
As before, SonarQube and SonarLint both carp if the construction used here is not used. I've done this to keep them happy and keep away from an issue